crates/server/src/datastore/postgres/mod.rs

```diff
@@ -0,0 +1,774 @@
+use self::models::{
+    NewCheckpoint, NewSource, ParsedText, RecordStatus, Source, SourceKind, Text, TextRef,
+};
+use super::{DataStore, DataStoreError, InitialLeaf, OperatorLogEntry, PackageLogEntry};
+use crate::datastore::postgres::models::{Checkpoint, NewLog, NewRecord};
+use anyhow::Result;
+use diesel::{prelude::*, result::DatabaseErrorKind};
+use diesel_async::{
+    pooled_connection::{deadpool::Pool, AsyncDieselConnectionManager},
+    scoped_futures::ScopedFutureExt,
+    AsyncConnection, AsyncPgConnection, RunQueryDsl,
+};
+use diesel_json::Json;
+use futures::{Stream, StreamExt};
+use serde::{de::DeserializeOwned, Serialize};
+use std::{collections::HashSet, pin::Pin};
+use warg_api::content::{ContentSource, ContentSourceKind};
+use warg_crypto::{
+    hash::DynHash,
+    signing::{KeyID, Signature},
+    Decode,
+};
+use warg_protocol::{
+    operator, package,
+    registry::{LogId, LogLeaf, MapCheckpoint, RecordId},
+    ProtoEnvelope, SerdeEnvelope,
+};
+
+mod models;
+mod schema;
+
+/// Trait for abstracting operator/package validator interactions for persistence.
+trait Validator: std::fmt::Debug + Serialize + DeserializeOwned + Default + Send + Sync {
```

Comment:
Should this be defined in the `warg-protocol` crate?

Reply:
Happy to move it there.